Updated patch for 3.6.15 to support Xcode 13.3 - as described at http…

nshine opened the pull request:
…s://bugs.python.org/issue45405

### Solve the following https://bugs.python.org/issue45405 

### Tests
-- pyenv install 3.6.15 (on latest MacOS version)
BEFORE:

python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 12.3 using python-build 2.2.5-4-g986fe1a7)

Inspect or clean up the working tree at /var/folders/n2/d8522dmx2sb2n6cqx1cks2140000gs/T/python-build.20220320191907.55294
Results logged to /var/folders/n2/d8522dmx2sb2n6cqx1cks2140000gs/T/python-build.20220320191907.55294.log

Last 10 log lines:
checking for --with-cxx-main=<compiler>... no
checking for clang++... no
configure:

  By default, distutils will build C++ extension modules with "clang++".
  If this is not intended, then set CXX on the configure command line.

checking for the platform triplet based on compiler characteristics... darwin
configure: error: internal configure error for the platform triplet, please file a bug report
make: *** No targets specified and no makefile found.  Stop.


AFTER:
Install success

native-api reviewed (changes requested):
Please make it as a separate patch so as not to mix up unrelated changes.

nshine commented:
Hi @native-api  - to generate this PR, I cloned CPython using the below and modified the configuration file from it, copy previous changes and added the change on it.
Do you want me to cherry pick only the specific change based the previous version ? (although the below seems more code accurate.)

git clone https://github.com/python/cpython --branch v3.6.15 --single-branch

native-api commented:
* Clone CPython and check out the appropriate tag
* Apply existing patches and commit (with either `patch` or `git am`)
* cherry-pick the upstream commit
* commit and `git format-patch`
* Enjoy!

nshine commented:
Hi @native-api  - please have a look now and see if this works for you

native-api reviewed (approved):
Perfect!
